org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/DeviceRequestTest.java

```diff
@@ -89,50 +102,55 @@ public void createServices() throws Exception {
 		info.setIcon("org.eclipse.scanning.example/icon/mandelbrot.png");
 		mandy.setDeviceInformation(info);
 		((RunnableDeviceServiceImpl)dservice)._register("mandelbrot", mandy);
-		
+
+		final DummyMalcolmDevice malc = new DummyMalcolmDevice();
+		final DeviceInformation<DummyMalcolmModel> malcInfo = new DeviceInformation<>();
+		malcInfo.setName("malcolm");
+		malcInfo.setLabel("Malcolm");
+		malcInfo.setDescription("Example malcolm device");
+		malcInfo.setId("org.eclipse.scanning.example.malcolm.dummyMalcolmDevice");
+		malc.setDeviceInformation(malcInfo);
+
+		((RunnableDeviceServiceImpl) dservice)._register("malcolm", malc);
 
 		Services.setRunnableDeviceService(dservice);
 		Services.setEventService(eservice);
-	
-		connect(eservice, dservice);
+
+		connect();
 	}
-	
+
 	@Before
-	public void start() throws Exception {
-		
+	public void start() {
+
 	   	Constants.setNotificationFrequency(200); // Normally 2000
 	   	Constants.setReceiveFrequency(100);
 	}
-	
+
 	@After
 	public void stop() throws Exception {
-		
+
     	Constants.setNotificationFrequency(2000); // Normally 2000
     	if (requester!=null) requester.disconnect();
     	if (responder!=null) responder.disconnect();
 	}
 
-	protected void connect(IEventService eservice, IRunnableDeviceService dservice) throws Exception {
-		
-		this.eservice = eservice;
-		this.dservice = dservice;
-		
+	protected void connect() throws Exception {
 		DeviceServlet dservlet = new DeviceServlet();
 		dservlet.setBroker(uri.toString());
-		dservlet.setRequestTopic(IEventService.DEVICE_REQUEST_TOPIC);
-		dservlet.setResponseTopic(IEventService.DEVICE_RESPONSE_TOPIC);
+		dservlet.setRequestTopic(EventConstants.DEVICE_REQUEST_TOPIC);
+		dservlet.setResponseTopic(EventConstants.DEVICE_RESPONSE_TOPIC);
 		dservlet.connect();
-				
+
 		// We use the long winded constructor because we need to pass in the connector.
-		// In production we would normally 
-		requester  = eservice.createRequestor(uri, IEventService.DEVICE_REQUEST_TOPIC, IEventService.DEVICE_RESPONSE_TOPIC);
-		requester.setTimeout(10, TimeUnit.SECONDS); // It's a test, give it a little longer.
+		// In production we would normally
+		requester  = eservice.createRequestor(uri, EventConstants.DEVICE_REQUEST_TOPIC, EventConstants.DEVICE_RESPONSE_TOPIC);
+		requester.setTimeout(10, TimeUnit.MINUTES); // It's a test, give it a little longer. // TODO change back to SECONDS
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)